pkg/workflows/upgrade.go

```diff
@@ -479,10 +526,15 @@ func (s *moveManagementToWorkloadTask) Name() string {
 }
 
 func (s *moveManagementToWorkloadTask) Checkpoint() *task.CompletedTask {
-	return nil
+	return &task.CompletedTask{
+		Checkpoint: nil,
+	}
 }
 
 func (s *moveManagementToWorkloadTask) Restore(ctx context.Context, commandContext *task.CommandContext, completedTask *task.CompletedTask) (task.Task, error) {
+	if !commandContext.ManagementCluster.ExistingManagement {
```

Comment:
When are we toggling this during the workflow if we change the management from the workload cluster? Or is this to mainly support the self-management cluster + workload cluster case?

Reply:
Yes this is to support the self-management cluster + workload cluster case